src/python/pants/build_graph/build_configuration.py

```diff
@@ -108,28 +110,46 @@ def _register_exposed_context_aware_object_factory(self, alias, context_aware_ob
 
     self._exposed_context_aware_object_factory_by_alias[alias] = context_aware_object_factory
 
+  @deprecated('1.15.0.dev1', hint_message='Use self.register_optionables().')
   def register_subsystems(self, subsystems):
+    return self.register_optionables(subsystems)
+
+  def register_optionables(self, optionables):
     """Registers the given subsystem types.
```

Comment:
Docstring should mention "optionable" now, not "subsystem"?

Reply:
Also, why is this bit necessary? What are the non-subsystem optionables we care about in this context?

Reply:
`Subsystem.scoped(cls)` results in a `SubsystemDependency`, which implements `OptionableFactory` in this PR. And in #6880, `Goal` implements `Optionable` rather than `Subsystem`.

Eventually we might be able to shrink the `Optionable`/`Subsystem` split... but this PR widens that gap a bit.